app/src/main/java/com/sdp/swiftwallet/common/FirebaseUtil.java

```diff
@@ -2,45 +2,50 @@
 
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.firestore.FirebaseFirestore;
-import com.sdp.cryptowalletapp.BuildConfig;
 
+/**
+ * Firebase authentication and Firestore related static methods
+ */
 public class FirebaseUtil {
-    /** Use emulators only in debug builds **/
+
+    // Use emulators only in debug builds
     private static final boolean sUseEmulators = false;
     private static final int FIREBASE_AUTH_EMULATOR_HOST = 9099;
     private static final int FIREBASE_FIRESTORE_EMULATOR_HOST = 8080;
+    // Special android domain for emulator to connect to the localhost
     private static final String ANDROID_HOST =  "10.0.2.2";
 
-
-
+    // Firebase Auth  and Firestore db
     private static FirebaseFirestore FIRESTORE;
     private static FirebaseAuth AUTH;
 
+    /**
+     * Gets the current used Firestore
+     * @return instance of Firestore if non null
+     */
     public static FirebaseFirestore getFirestore() {
         if (FIRESTORE == null) {
             FIRESTORE = FirebaseFirestore.getInstance();
-
-            // Connect to the Cloud Firestore emulator when appropriate. The host '10.0.2.2' is a
-            // special IP address to let the Android emulator connect to 'localhost'.
             if (sUseEmulators) {
-                FIRESTORE.useEmulator("10.0.2.2", FIREBASE_FIRESTORE_EMULATOR_HOST);
+                FIRESTORE.useEmulator(ANDROID_HOST, FIREBASE_FIRESTORE_EMULATOR_HOST);
             }
         }
-
         return FIRESTORE;
     }
 
+    /**
+     * Gets the current Firebase Authentication object
+     * @return the current Firebase Authentication if non null
+     */
     public static FirebaseAuth getAuth() {
-        if (AUTH == null) {
-            AUTH = FirebaseAuth.getInstance();
 
-            // Connect to the Firebase Auth emulator when appropriate. The host '10.0.2.2' is a
-            // special IP address to let the Android emulator connect to 'localhost'.
+        if (AUTH != null) {
```

Comment:
I think you messed up here, it would be better to leave it as it was and only change the comments and doc in this file.

Reply:
its ok purpose, i I don’t add it codeclimate is gonna complain about code duplication